### PR TITLE
chore(query): improve filter string kernel

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
How much performance has been improved?

**Comment by sundy-li** [comment]
> How much performance has been improved?

It's just little better

File: src/query/expression/src/kernels/filter.rs [hunk 1]
```diff
@@ -317,4 +311,87 @@ impl Column {
...
+        let mut res_offsets = Vec::with_capacity(selected + 1);
+        res_offsets.push(0);
+
+        let mut res_data = vec![];
+        let hint_size = data.len() / (values.len() + 1) * selected;
+
+        static MAX_HINT_SIZE: usize = 1000000000;
+        if hint_size < MAX_HINT_SIZE && values.len() < MAX_HINT_SIZE {
+            res_data.reserve(hint_size)
+        }
```
Comment (RinChanNOWWW) [inline src/query/expression/src/kernels/filter.rs:334]: Why limit the hint size?

Comment (sundy-li) [reply src/query/expression/src/kernels/filter.rs:334]: it's hint size, avoid hint too much capacity.

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**